src/main/java/com/opinta/controller/DiscountPerCounterpartyController.java

```diff
@@ -0,0 +1,114 @@
+import static com.opinta.util.LogMessageUtil.deleteOnErrorLogEndpoint;
+import static com.opinta.util.LogMessageUtil.getAllOnErrorLogEndpoint;
+import static com.opinta.util.LogMessageUtil.getByIdOnErrorLogEndpoint;
+import static com.opinta.util.LogMessageUtil.saveOnErrorLogEndpoint;
+import static com.opinta.util.LogMessageUtil.updateOnErrorLogEndpoint;
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+import static org.springframework.http.HttpStatus.OK;
+import static org.springframework.http.HttpStatus.UNAUTHORIZED;
+
+@RestController
+@RequestMapping("/discounts-per-counterparty")
+public class DiscountPerCounterpartyController {
+    private final DiscountPerCounterpartyService discountPerCounterpartyService;
+    private final UserService userService;
+    
+    @Autowired
+    public DiscountPerCounterpartyController(DiscountPerCounterpartyService discountPerCounterpartyService,
+                                             UserService userService) {
+        this.discountPerCounterpartyService = discountPerCounterpartyService;
+        this.userService = userService;
+    }
+    
+    @GetMapping
+    public ResponseEntity<?> getAllDiscountsPerCounterparty(@RequestParam UUID token) {
+        try {
+            User user = userService.authenticate(token);
+            return new ResponseEntity<>(discountPerCounterpartyService.getAll(user), OK);
+        } catch (AuthException e) {
+            return new ResponseEntity<>(getAllOnErrorLogEndpoint(DiscountPerCounterparty.class, e), UNAUTHORIZED);
+        }
+    }
+    
+    @GetMapping("{uuid}")
+    public ResponseEntity<?> getDiscountPerCounterpartyByUuid(@PathVariable UUID uuid, @RequestParam UUID token) {
+        try {
+            User user = userService.authenticate(token);
+            return new ResponseEntity<>(discountPerCounterpartyService.getByUuid(uuid, user), OK);
+        } catch (AuthException e) {
+            return new ResponseEntity<>(authorizationOnErrorLogEndpoint(token, e), UNAUTHORIZED);
+        } catch (IncorrectInputDataException e) {
+            return new ResponseEntity<>(getByIdOnErrorLogEndpoint(DiscountPerCounterparty.class, e), NOT_FOUND);
+        }
+    }
+    
+    @PostMapping
+    public ResponseEntity<?> createDiscountPerCounterparty(
+            @RequestBody DiscountPerCounterpartyDto discountPerCounterpartyDto, @RequestParam UUID token) {
+        try {
+            User user = userService.authenticate(token);
+            return new ResponseEntity<>(discountPerCounterpartyService.save(discountPerCounterpartyDto, user), OK);
+        } catch (AuthException e) {
+            return new ResponseEntity<>(authorizationOnErrorLogEndpoint(token, e), UNAUTHORIZED);
+        } catch (IncorrectInputDataException e) {
+            return new ResponseEntity<>(saveOnErrorLogEndpoint(DiscountPerCounterparty.class, e), NOT_FOUND);
```

Comment:
Plz, change to saveOnErrorLogEndpoint(DiscountPerCounterparty.class, discountPerCounterpartyDto, e)